spark/datadog_checks/spark/spark.py

```diff
@@ -63,92 +63,90 @@
-    'totalTasks': ('spark.executor.total_tasks', INCREMENT),
-    'totalDuration': ('spark.executor.total_duration', INCREMENT),
-    'totalInputBytes': ('spark.executor.total_input_bytes', INCREMENT),
-    'totalShuffleRead': ('spark.executor.total_shuffle_read', INCREMENT),
-    'totalShuffleWrite': ('spark.executor.total_shuffle_write', INCREMENT),
-    'maxMemory': ('spark.executor.max_memory', INCREMENT)
+    'rddBlocks': ('spark.executor.rdd_blocks', COUNT),
+    'memoryUsed': ('spark.executor.memory_used', COUNT),
+    'diskUsed': ('spark.executor.disk_used', COUNT),
+    'activeTasks': ('spark.executor.active_tasks', COUNT),
+    'failedTasks': ('spark.executor.failed_tasks', COUNT),
+    'completedTasks': ('spark.executor.completed_tasks', COUNT),
+    'totalTasks': ('spark.executor.total_tasks', COUNT),
+    'totalDuration': ('spark.executor.total_duration', COUNT),
+    'totalInputBytes': ('spark.executor.total_input_bytes', COUNT),
+    'totalShuffleRead': ('spark.executor.total_shuffle_read', COUNT),
+    'totalShuffleWrite': ('spark.executor.total_shuffle_write', COUNT),
+    'maxMemory': ('spark.executor.max_memory', COUNT)
 }
 
 SPARK_RDD_METRICS = {
-    'numPartitions': ('spark.rdd.num_partitions', INCREMENT),
-    'numCachedPartitions': ('spark.rdd.num_cached_partitions', INCREMENT),
-    'memoryUsed': ('spark.rdd.memory_used', INCREMENT),
-    'diskUsed': ('spark.rdd.disk_used', INCREMENT)
+    'numPartitions': ('spark.rdd.num_partitions', COUNT),
+    'numCachedPartitions': ('spark.rdd.num_cached_partitions', COUNT),
+    'memoryUsed': ('spark.rdd.memory_used', COUNT),
+    'diskUsed': ('spark.rdd.disk_used', COUNT)
 }
 
 SPARK_STREAMING_STATISTICS_METRICS = {
-    'avgInputRate': ('spark.streaming.statistics.avg_input_rate', GAUGE),
-    'avgProcessingTime': ('spark.streaming.statistics.avg_processing_time', GAUGE),
-    'avgSchedulingDelay': ('spark.streaming.statistics.avg_scheduling_delay', GAUGE),
-    'avgTotalDelay': ('spark.streaming.statistics.avg_total_delay', GAUGE),
-    'batchDuration': ('spark.streaming.statistics.batch_duration', GAUGE),
-    'numActiveBatches': ('spark.streaming.statistics.num_active_batches', GAUGE),
-    'numActiveReceivers': ('spark.streaming.statistics.num_active_receivers', GAUGE),
-    'numInactiveReceivers': ('spark.streaming.statistics.num_inactive_receivers', GAUGE),
-    'numProcessedRecords': ('spark.streaming.statistics.num_processed_records', MONOTONIC_COUNT),
-    'numReceivedRecords': ('spark.streaming.statistics.num_received_records', MONOTONIC_COUNT),
-    'numReceivers': ('spark.streaming.statistics.num_receivers', GAUGE),
-    'numRetainedCompletedBatches': ('spark.streaming.statistics.num_retained_completed_batches', MONOTONIC_COUNT),
-    'numTotalCompletedBatches': ('spark.streaming.statistics.num_total_completed_batches', MONOTONIC_COUNT)
+    'avgInputRate': ('spark.streaming.statistics.avg_input_rate', COUNT),
```

Comment:
@ofek I'm not sure about the Spark streaming metrics, we're not using Spark streaming at Datadog so I didn't look at them. Looking at the check, it seems these metrics are only per app and we tag them by app_id so there is no aggregation issue as for the other metrics. @gzussa Seems like you added it, how did you test it?

Reply:
Indeed you can revert those changes. It has been tested and vetted by some customers already.

Reply:
Done!